matrix-org/matrix-react-sdk
# Improvements to alt aliases

Fixes: https://github.com/vector-im/riot-web/issues/13501

## Testing notes

* Dial up the latency on your web browser to 4000ms or so to check it spinners properly
* Check with normal latency, it should feel snappy
* Add an alias on another server, get an error dialog
* Add an invalid alias such as `test`.  We don't locally validate (yet) but the feedback appears on the `Field` element as if we did
* Add an alias that doesn't point to the room, such as `#test:localhost` etc. Feedback should appear on the `Field` element too
* Add a bad alias on another server such as `#test:example`. Feedback should come as a dialog box for now.

## Visual changes

* Aliases dialog now also has a prefixed `#` and allows input of aliases either beginning or not beginning with `#`.
* Failing to add an alias leaves the user's input in the input field, so they can correct small typos
* Failing to add an alias does not add it to the list of aliases (this was a bug)

**Inline feedback for room aliases which don't point to the room**
This replaces a dialog reading "There was an error updating the room's alternative addresses. It may not be allowed by the server or a temporary failure occurred."
<img width="819" alt="Screenshot 2020-05-12 at 15 10 27" src="https://user-images.githubusercontent.com/3935731/81702190-dd604080-9462-11ea-842b-8f487521f5ac.png">

**Inline feedback for bad aliases**
This replaces a dialog reading "There was an error updating the room's alternative addresses. It may not be allowed by the server or a temporary failure occurred."
<img width="808" alt="Screenshot 2020-05-12 at 15 10 43" src="https://user-images.githubusercontent.com/3935731/81702200-e05b3100-9462-11ea-8bd1-3db74316898a.png">

**Overlay entire component while busy**
This is because, for technical reasons, all of these controls back on to one network call.
<img width="746" alt="Screenshot 2020-05-12 at 15 11 02" src="https://user-images.githubusercontent.com/3935731/81702409-23b59f80-9463-11ea-8b7d-f0c5bf8b0eb6.png">

## Video

[Thread](https://matrix.to/#/!ObFGQSDjOOQNXJntbN:matrix.org/$C1JfKA7ZXXhkoblnVXHdhBlxHZlyDKeKDRNZzecuPOI?via=matrix.org)

[Video](https://matrix-client.matrix.org/_matrix/media/r0/download/matrix.org/JshrJQteaLLLXHMzbTPMjqZN)
